plugins/se/se.cpp

```diff
@@ -3,8 +3,20 @@
 // that can be found in the LICENSE file at the root of the
 // Mumble source tree or at <https://www.mumble.info/LICENSE>.
 
-#include "../mumble_plugin_main.h"
-#include "../mumble_plugin_utils.h"
+#include "../mumble_plugin.h"
+
+#ifdef OS_LINUX
+#	include "../MumblePluginProcessLinux.hpp"
+#endif
+#include "../MumblePluginProcessWindows.hpp"
+
+std::unique_ptr< MumblePluginProcess > proc;
+
+#ifdef OS_WINDOWS
+constexpr static bool isWin32 = true;
+#else
+static bool isWin32;
+#endif
```

Comment:
1. Why is one labeled `constexpr` and the other `static`?
2. I would explicitly initialize `isWin32 = false` as well. Relying on default values is bit smelly imo.

Reply:
1. Both are labeled `static`. The reason why it's `constexpr` on Windows is because the process is guaranteed to be native.
The variable is used in the plugin's headers to determine how to parse some specific parts of the game's assembly.
2. `static` global variables are actually initialized to 0 by default: https://stackoverflow.com/questions/2091499/why-are-global-and-static-variables-initialized-to-their-default-values

Reply:
Oh yes indeed - I always use `static` as the first keyword wherever I am using it. That's why I missed it ^^

I'm not sure if I like the type of a variable to be dependent on the OS this is compiled on. I think in this case I would opt for removing `constexpr` altogether. I do see that there are also benefits to not doing so :shrug: 

Yeah I know that they are initialized to defaults but it makes reading the code slightly more complicated if you have to check whether in this current situation the variable will have a proper default initialization and what that will end up being. Therefore I prefer explicitly initializing such variables in code, so that it is obvious what the value of this variable will be :point_up:

Reply:
I agree, I will explicitly initialize `isWin32` to `false`.

As for `constexpr`: it tells the compiler that the variable's value is never going to change.
However, I think in this specific case `const` achieves the same behavior.

Reply:
I know what it does. And `const` would be just as "bad" as `constexpr` in this case. My problem here is that one time you have a `static bool` and anohter time you have the type `const(expr) static bool` :shrug:

Reply:
I don't understand what the problem is though. The data type is still `bool`.

Reply:
No it's not. The const-qualifiers and the `static` are also part of the type. So we have the type `static bool` vs `constexpr static bool` for one and the same variable depending on the platform this is compiled on.
Imo this is not a good design as you potentially have to use other preprocessor statements again when working with this variable in order to differentiate the cases.

A variable's type should not change depending on the platform (imo).